infrastructure/src/main/java/org/corfudb/infrastructure/NettyServerRouter.java

```diff
@@ -97,39 +96,32 @@ protected static void handleUncaughtException(Thread t, @Nonnull Throwable e) {
     @Setter
     long serverEpoch;
 
-    /**
-     * Returns a new NettyServerRouter.
-     * @param opts map of options (FIXME: unused)
-     */
-    public NettyServerRouter(Map<String, Object> opts) {
-        handlerMap = new ConcurrentHashMap<>();
-    }
+    /** The {@link AbstractServer}s this {@link NettyServerRouter} routes messages for. */
+    @Getter
+    final List<AbstractServer> servers;
 
-    /**
-     * Add a new netty server handler to the router.
+    /** Construct an new {@link NettyServerRouter}.
      *
-     * @param server The server to add.
+     * @param servers   A list of {@link AbstractServer}s this router will route
+     *                  messages for.
      */
-    public void addServer(AbstractServer server) {
-        // Iterate through all types of CorfuMsgType, registering the handler
-        server.getHandler().getHandledTypes()
-                .forEach(x -> {
-                    handlerMap.put(x, server);
-                    log.trace("Registered {} to handle messages of type {}", server, x);
-                });
+    public NettyServerRouter(List<AbstractServer> servers) {
+        this.servers = servers;
+        handlerMap = new EnumMap<>(CorfuMsgType.class);
+        servers.forEach(server -> server.getHandler().getHandledTypes()
+            .forEach(x -> handlerMap.put(x, server)));
     }
 
     /**
-     * Remove a server from the router.
-     * @param server  server to remove
+     * {@inheritDoc}
+     *
+     * @deprecated This operation is no longer supported. The router will only route messages for
+     * servers provided at construction time.
      */
-    public void removeServer(AbstractServer server) {
-        // Iterate through all types of CorfuMsgType, un-registering the handler
-        server.getHandler().getHandledTypes()
-                .forEach(x -> {
-                    handlerMap.remove(x, server);
-                    log.trace("Un-Registered {} to handle messages of type {}", server, x);
-                });
+    @Override
+    @Deprecated
+    public void addServer(AbstractServer server) {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)